### PR TITLE
Dialyzer

### PR DESCRIPTION
**Pull request by eksperimental**
Use Dialyzer for testing locally ~and in the CI~

### CODE REVIEW

**Comment by josevalim** [comment]
Thank you but I am a :-1:. I don't think Dialyzer is worth the ROI. If someone wants to run it locally and provide fixes, that's welcome, but its error messages are hard to follow up to make it part of CI.

**Comment by eksperimental** [comment]
Maybe I can install something like https://github.com/michallepicki/elixir-lang-dialyzer-runs and keep an eye on it.
I will remove the CI part then

**Comment by eksperimental** [comment]
done

File: mix.exs [hunk 1]
```diff
@@ -35,7 +40,8 @@ defmodule ExDoc.Mixfile do
       {:makeup_elixir, "~> 0.14"},
       {:makeup_erlang, "~> 0.1"},
       {:excoveralls, "~> 0.3", only: :test},
-      {:jason, "~> 1.2", only: :test}
+      {:jason, "~> 1.2", only: :test},
+      {:dialyxir, "~> 1.0", only: [:dev, :test], runtime: false}
```
Comment (josevalim) [inline mix.exs:44]:
```suggestion
      {:dialyxir, "~> 1.0", only: [:dialyzer], runtime: false}
```

**Review by josevalim — APPROVED** [review]

File: mix.exs [hunk 2]
```diff
@@ -35,7 +39,8 @@ defmodule ExDoc.Mixfile do
       {:makeup_elixir, "~> 0.14"},
       {:makeup_erlang, "~> 0.1"},
       {:excoveralls, "~> 0.3", only: :test},
-      {:jason, "~> 1.2", only: :test}
+      {:jason, "~> 1.2", only: :test},
+      {:dialyxir, "~> 1.0", only: [:dialyzer], runtime: false}
```
Comment (josevalim) [inline mix.exs:43]:
```suggestion
      {:dialyxir, "~> 1.0", only: :dialyzer, runtime: false}
```

**Review by josevalim — APPROVED** [review]

**Comment by eksperimental** [comment]
Thank you @josevalim
